main.go

```diff
@@ -109,6 +115,52 @@ func newRootCommand() *cobra.Command {
 	return cmd
 }
 
+func sleepUntilTERM() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, os.Interrupt, syscall.SIGTERM)
+
+	for {
+		select {
+		case sig := <-sigs:
+			if sig == syscall.SIGTERM {
+				signal.Stop(sigs)
+				klog.Info("SIGTERM caught")
+				break
+			}
+		}
+	}
+}
+
+func detectOCP411(cfg *rest.Config) {
+	// Detect cluster version
+	// and cease operation (do not die to avoid respawn) when on 4.11
+	// as the PAO logic is provided by NTO there
+	k8sclient, err := client.New(cfg, client.Options{})
+	if err != nil {
+		klog.Fatalf("could not detect cluster version: %v", err)
+	}
+
+	// Get OCP version objects (there should be just one)
+	clusterversionlist := v1.ClusterVersionList{}
+	ctx := context.Background()
+	err = k8sclient.List(ctx, &clusterversionlist)
```

Comment:
nit: some syntax sugar
```
if err := k8sclient.List(context.TODO(), &clusterversionlist); err != nil {
    ...
}
```

Reply:
Line too long :D But yeah, I can do that.